app/services/account_reset/delete_account.rb

```diff
@@ -39,6 +39,17 @@ def track_mfa_method_counts
       @mfa_method_counts = MfaContext.new(user).enabled_two_factor_configuration_counts_hash
     end
 
+    def verification_method
+      return nil if user.uuid == 'anonymous-uuid'
```

Comment:
Why would we ever have an anonymous user?

Reply:
That looks like the scenario where an invalid token is provided. This class looks up a user by mapping the given token to a deletion request.

I might have expected that `user` would be `nil` for an invalid token, but it seems like this was implemented to use the `AnonymousUser` instead. If we want to lean into that, and `AnonymousUser` is supposed to behave like a real user instance, maybe we can implement whatever missing methods are needed.

Or alternatively, since `AnonymousUser` does implement `identity_verified?`, and we'd only expect a result from this method if the user is verified, we can bail on that. A bonus is that we avoid additional database queries for pending in-person enrollment and GPO verification pending profile.

```suggestion
      return nil if !user.identity_verified?
```

Reply:
I see. I was under a misunderstanding that we'd want to log pending verified deletions also.
Rereading the slack thread I don't see anything regarding a pending user deletion.

Reply:
Yeah, actually now that you mention it, we're interested in verified identities, so `pending_in_person_enrollment` and `gpo_verification_pending_profile` are not what we want to check.